main.py

```diff
@@ -142,12 +241,34 @@ def api_template(my_path=''):
         if UPLOAD_LIMIT is not None and CURRENT_UPLOAD_SIZE >= UPLOAD_LIMIT:
             return Response(status=400, response='Reached upload limit')
         with open(os.path.join('static/data/', my_path,  'template.json'), "w+") as outfile:
+            data = request.get_json()
+        # Call the validate function from jsonschema to verify our output json conforms with the template we were supposed to create
+            try:
+                validate(data, TEMPLATE_SCHEMA)
+            except:
+                return Response(status=400, response='Invalid template, requires at least 1 label and proper formatting. Refer to schema.')
             json.dump(request.get_json(), outfile, sort_keys=True, indent=4)
+        # Open template file to make sure what we just created matches what was submitted. In theory we should never get an except here considering we JUST wrote the file. 
+        try:
+            template = open(os.path.join('static/data/',(my_path + 'template.json')))
+            # Convert the template json to something validate can understand.
+            template = json.load(template)
+        except:
+            return Response(status=400, response='Unable to find template.')
+
         if UPLOAD_LIMIT is not None:
             CURRENT_UPLOAD_SIZE = CURRENT_UPLOAD_SIZE + 1
         return json.dumps({'success': True}), 200, {'ContentType': 'application/json'}
 
-
+def fetch_template(my_path):
+    iterations = 0
+    while not os.path.isfile(os.path.join(my_path, 'template.json')):
+            my_path = os.path.normpath(os.path.join(my_path, '../'))
+            # print(my_path)
+            iterations = iterations + 1
+            if my_path== 'static' or iterations == 500:
+                return Response(status=404, response='Unable to find parent template')
```

Comment:
cant return a response because this isn't a route. should return None then the function that call it should handle this potential None